### PR TITLE
Fixed all comments

### PR DESCRIPTION
**Pull request by harmeshverma01**

### CODE REVIEW

File: cart/views.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+from webbrowser import get
```
Comment (pythexcel) [inline cart/views.py:1]: Please remove unwanted imports.

File: cart/views.py [hunk 2]
```diff
@@ -13,8 +16,8 @@ class CartView(APIView):
     authentication_classes = [authentication.TokenAuthentication]
```
Comment (pythexcel) [inline cart/views.py:16]: Please explain why we are using permission_classes IsAuthenticated and authentication_classes together .

File: cart/views.py (hunk 1 above)
Comment (harmeshverma01) [reply cart/views.py:1]: fixed

File: cart/views.py (hunk 2 above)
Comment (harmeshverma01) [reply cart/views.py:16]: fixed

**Review by pythexcel — COMMENTED** [review]
class Userdetailsview(APIView)
Do we need admin required in this view. Please check and fix.

**Review by pythexcel — CHANGES_REQUESTED** [review]
class UserProfile(APIView):
    serializer_class = ProfileSerializer
    
    def get(self, request, id=None):
        profile = Profile.objects.all()
        serializer = self.serializer_class(profile, many=True)
        return Response(serializer.data)
        

-        Is this correct here ? Any one can fetch all user's data with this. 

 class LoginView(APIView):
    serializer_class = UserSerializer
    
    def post(self, request):
        user = authenticate(email=request.data.get('email'), password=request.data.get('password'))
        if user:
            token = Token.objects.get_or_create(user=user)
            return Response({'token': str(token[0])})
        return Response({'details': 'User Not Found'}, status=status.HTTP_404_NOT_FOUND)
        

-       What message user will get in case if he/she entred  wrong password  ? please fix. 

class RagisterView(APIView):
    serializer_class = UserSerializer
    
    def post(self, request):
        serializer = self.serializer_class(data=request.data)
        user = None
        if serializer.is_valid():
            user = serializer.save(password=make_password(request.data['password']))
            if user:
                token = Token.objects.get_or_create(user=user)
            return Response({'token': str(token[0])})
        return Response(({'details':'Something went wrong!'}), status=status.HTTP_404_NOT_FOUND)
        

-         why we are not showing error using serializer ?

class CreateprofileView(APIView):
    serializer_class = ProfileSerializer
    authentication_classes = [authentication.TokenAuthentication]
    permission_classes = [permissions.IsAuthenticated]
    
    def post(self, request):
        serializer = self.serializer_class(data=request.data)
        if serializer.is_valid():
            serializer.save()
        return Response(serializer.data, status=status.HTTP_201_CREATED)
        

-         Is this condition correct here ? Please check and update for all .

**Review by pythexcel — COMMENTED** [review]
def delete(self, request, id=None):
        order = Order.objects.get(id=id)
        order.delete()
        return Response(({'message': 'order is deleted'}), status=status.HTTP_204_NO_CONTENT) 

-         error handling is not proper

class StoreView(APIView):
    serializer_class = StoreSerializer
    permission_classes = [permissions.IsAuthenticated, both_required]
    authentication_classes = [authentication.TokenAuthentication]

    def get(self, request, id=None):
        store = Store.objects.all()
        serializer = self.serializer_class(store, many=True)
        return Response(serializer.data)

-         logic need to update here.

def patch(self, request, id=None):
        try:
            user = User.objects.get(id=id)
            serializer = self.serializer_class(user, data=request.data, partial=True)
            if serializer.is_valid():
                serializer.save()
            return Response(serializer.data, status=status.HTTP_202_ACCEPTED)
        except:
            return Response(({"message": "User not Found"}), status=status.HTTP_404_NOT_FOUND)
            

-             try to find defination for status-code 202 and let me know if need to use this here ir not. 


 def patch(self, request, id=None):
        try:
            if request.user.id == id:
                user = Profile.objects.get(id=id)
                serializer = self.serializer_class(user, data=request.data, partial=True)
                return Response(serializer.data, status=status.HTTP_200_OK)
            else:
                return Response(({'UnAuthorized User'}), status=status.HTTP_401_UNAUTHORIZED)
        except:
            return Response( status=status.HTTP_404_NOT_FOUND)  
            

-             what is the use of this  Response(({'UnAuthorized User'}), status=status.HTTP_401_UNAUTHORIZED).

**Pull request merged**